.travis.yml

```diff
@@ -1,8 +1,55 @@
+sudo: false
+
 language: php
-php:  
-  - 5.6
-  - 7.0
 
-before_script: composer install
+cache:
+  directories:
+    - $HOME/.composer/cache
+
+env:
+  global:
+    - COMPOSER_ARGS="--no-interaction"
+    - LEGACY_DEPS="phpunit/phpunit"
+
+matrix:
+  include:
+    - php: 5.6
+      env:
+        - DEPS=lowest
+    - php: 5.6
+      env:
+        - DEPS=latest
+    - php: 7
+      env:
+        - DEPS=lowest
+    - php: 7
+      env:
+        - DEPS=latest
+    - php: 7.1
+      env:
+        - DEPS=lowest
+    - php: 7.1
+      env:
+        - DEPS=latest
+    - php: 7.2
+      env:
+        - DEPS=lowest
+    - php: 7.2
+      env:
+        - DEPS=latest
+  allow_failures:
+    - php: 7.2
+
+before_install:
+  - travis_retry composer self-update
+
+install:
+  - if [[ $TRAVIS_PHP_VERSION =~ ^5.6 ]]; then travis_retry composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
+  - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
+  - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
+
+script:
+  - composer test
 
-script: vendor/bin/phpunit
+notifications:
+  email: false
```

Comment:
Let them notify failures or success. Are you getting too much of notifications ? :) .

Reply:
I dont get any... :-)

Reply:
then lets remove this.